api/v1alpha1/releaseplanadmission_types.go

```diff
@@ -85,6 +103,23 @@ type ReleasePlanAdmission struct {
 	Status ReleasePlanAdmissionStatus `json:"status,omitempty"`
 }
 
+// ClearMatchingInfo marks the ReleasePlanAdmission as no longer matched to any ReleasePlan.
+func (rpa *ReleasePlanAdmission) ClearMatchingInfo() {
+	rpa.Status.ReleasePlans = []MatchedReleasePlan{}
+	conditions.SetCondition(&rpa.Status.Conditions, MatchedConditionType, metav1.ConditionFalse, MatchedReason)
+}
+
+// MarkMatched marks the ReleasePlanAdmission as matched to a given ReleasePlan.
+func (rpa *ReleasePlanAdmission) MarkMatched(releasePlan *ReleasePlan) {
+	pairedReleasePlan := MatchedReleasePlan{
+		Name:   fmt.Sprintf("%s%c%s", releasePlan.GetNamespace(), types.Separator, releasePlan.GetName()),
+		Active: (releasePlan.GetLabels()[metadata.AutoReleaseLabel] == "true"),
+	}
+	rpa.Status.ReleasePlans = append(rpa.Status.ReleasePlans, pairedReleasePlan)
```

Comment:
Thinking about this... maybe we need a sort in here. That way we would have less reconciliations when the list generated is exactly the same as it was. And it would be easier to read.

Reply:
Thinking a bit more on this... If called with the same ReleasePlan, the information would be duplicated. Shouldn't this be a map maybe?